openmc/mgxs/mgxs.py

```diff
@@ -675,14 +683,20 @@ def load_from_statepoint(self, statepoint):
             self.domain = statepoint.summary.get_universe_by_id(self.domain.id)
         elif self.domain_type == 'material':
             self.domain = statepoint.summary.get_material_by_id(self.domain.id)
+        elif self.domain_type == 'mesh':
+            self.domain = statepoint.meshes[self.domain.id]
         else:
             msg = 'Unable to load data from a statepoint for domain type {0} ' \
                   'which is not yet supported'.format(self.domain_type)
             raise ValueError(msg)
 
         # Use tally "slicing" to ensure that tallies correspond to our domain
         # NOTE: This is important if tally merging was used
-        if self.domain_type != 'distribcell':
+        if self.domain_type == 'mesh':
+            filters = [self.domain_type]
+            xyz = [np.arange(1, x+1) for x in self.domain.dimension]
+            filter_bins = [tuple(itertools.product(*xyz))]
```

Comment:
The numpy way to do this would be:
```Python
list(np.broadcast(*np.ogrid[1:x+1, 1:y+1, 1:z+1]))
```
Unfortunately you'd have to have separate cases for 2D and 3D, so in that sense I like yours better (I also think the meaning is clearer). However, I would suggest using just normal `range` instead of `np.arange` since at the end of the day you're getting a list of tuples.